src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/CreditManager.cs

```diff
@@ -120,7 +121,9 @@ public void Dispose()
                 {
                     while (_waiters.TryDequeue(out Waiter waiter))
                     {
-                        waiter.TrySetException(CreateObjectDisposedException(forActiveWaiter: true));
+                        waiter.TrySetException(abortException != null ? (Exception)
```

Comment:
Is the cast to `(Exception)` necessary?

Reply:
It is.  The compiler needs to pick a type to use for both sides of the ternary, and if both sides aren't the same or one side isn't convertible to the other, it errors.